source/class/qx/ui/form/FileSelectorButton.js

```diff
@@ -131,12 +141,13 @@ qx.Class.define("qx.ui.form.FileSelectorButton", {
     },
 
     _createContentElement: function () {
-      let id = "qxFileSelector_" + InputElementIdCounter++;
+      let id = qx.ui.form.FileSelectorButton._generateElementId();
       let input = (this.__inputObject = new qx.html.Input(
         "file",
-        { display: "none" },
+        null,
```

Comment:
Why do you prefer to have the element hidden after initially created, rather than initially?

Reply:
This is actually a behavior change. In the original case, it would consume no space, equivalent to a call to `this.exclude()`. `this.hide()` leaves the layout as if the widget were shown, but doesn't show the widget.

Reply:
It's not a preference, actually it doesnt work on my Chrome 102.0.5005.61 64bits linux using display none as style parameter on instantiation, still showing and I just dint deep dive on it.

I'll trye using exclude and not hide.

Reply:
Yes, using exlude works too, I just looking qoodoo for the first time, so I dont know all its nuances.

Reply:
Actually I found some problems with this, if exluded or hidden before first appear event (creation I think) it just doesnt create the dom element, and if using exclude during appear event the file dialog doesnt appear

Reply:
I'll await @oetiker's comments, as he implemented this recently. I think, though, that there is likely some other issue, as providing attributes "should" work. (I know, theory vs. reality, but still, it is supposed to work...)

BTW, I just noticed that `InputElementIdCounter` wasn't global. It was declared private to the file, with `let InputElementIdCounter = 0;`, which is legitimate. Your change makes it more "traditional qooxdoo"-like, but it wasn't polluting the global namespace.

@oetiker During your work on this, did you encounter any problems in Chrome (specifically, I guess, Chrome 102.0.5005.61)?

Reply:
Yes there was some problems, it still showing the input field like in this print taken using master branch.

![image](https://user-images.githubusercontent.com/1051974/183304866-33034287-809c-4e84-9e88-2b3ea6178a3f.png)

Reply:
And you are right, using let in file scope, now its up to you what implementation to use :)

Reply:
@derrell there could well be issues with chrome 102 ... I have not tested this

Reply:
The whole 'dom element does not get created if 'display' is set to 'none', would indicate that there is some optimization on the qx side at work which we should get to the bottom of I guess. @evandropoa is this whole thing really 'chrome dependent' ?

Reply:
It's not a Chrome only thing, the same happens on my FireFox 95.0.2 (64-bits) the imput element is rendered over the button when using the master branch.

About doesnt being created when **display** is set to **none** could be an optimization 4 sure, I just didnt deep dive on element creation yet.

Reply:
yes this has nothing todo with chrome ...